File: FishTrackerLambda/src/FishTrackerLambda.Services/Http/Result.cs
namespace FishTrackerLambda.Services.Http
{
    public class Results : IResult
    {
        public int StatusCode { get; private set; }
        public string Message { get; private set; }
        public object? Object { get; private set; }

        private Results(int statusCode, string message)
        {
            StatusCode = statusCode;
            Message = message;
            Object = null;
        }

        private Results(int statusCode, object? obj)
        {
            StatusCode = statusCode;
            Message = string.Empty;
            Object = obj;
        }

        public static IResult NotFound(string message = "Not Found")
        {
            return new Results(404, message);
        }

        public static IResult BadRequest(string message = "Bad Request")
        {
            return new Results(400, message);
        }

        public static IResult Ok(object? obj = null)
        {
            return new Results(200, obj);
        }

        public static IResult InternalServerError(string message = "Internal Server Error")
        {
            return new Results(500, message);
        }

        public static IResult Forbidden(string message = "Forbidden")
        {
            return new Results(403, message);
        }

        public static IResult Gone(string message = "Gone")
        {
            return new Results(410, message);
        }

        public static IResult PayloadTooLarge(string message = "Payload Too Large")
        {
            return new Results(413, message);
        }

        public static IResult TooManyRequests(string message = "Too Many Requests")
        {
            return new Results(429, message);
        }

        public static IResult StatusCodeResult(int statusCode, string message = "")
        {
            return new Results(statusCode, message);
        }
    }
}

File: FishTrackerLambdaNode/src/Http/Result.ts
//import { IResult } from './IResult';

// export interface IResult {
//     statusCode: number;
//     message: string;
//     object?: any;
// }

export class Results {
    statusCode: number;
    message: string;
    object?: any;

    private constructor(statusCode: number, message: string);
    private constructor(statusCode: number, obj: any);
    private constructor(statusCode: number, param: string | any) {
        this.statusCode = statusCode;
        if (typeof param === 'string') {
            this.message = param;
            this.object = null;
        } else {
            this.message = '';
            this.object = param;
        }
    }

    static NotFound(message: string = "Not Found"): Results {
        return new Results(404, message);
    }

    static BadRequest(message: string = "Bad Request"): Results {
        return new Results(400, message);
    }

    static Ok(obj: any = null): Results {
        return new Results(200, obj);
    }

    static Forbidden(message: string = "Forbidden"): Results {
        return new Results(403, message);
    }

    static Gone(message: string = "Gone"): Results {
        return new Results(410, message);
    }

    static PayloadTooLarge(message: string = "Payload Too Large"): Results {
        return new Results(413, message);
    }

    static TooManyRequests(message: string = "Too Many Requests"): Results {
        return new Results(429, message);
    }

    static InternalServerError(message: string = "Internal Server Error"): Results {
        return new Results(500, message);
    }

    static StatusCode(code: number, message: string = ""): Results {
        return new Results(code, message);
    }
}